mods-unpacked/ITR-SmarterDealer/bruteforce.gd

```diff
@@ -453,26 +460,20 @@ static func sum_array(array)->float:
 
 static func GetBestChoiceAndDamage_Internal(roundType: int, liveCount: int, blankCount: int, liveCount_max: int, player: BruteforcePlayer, opponent: BruteforcePlayer, tempStates: TempStates, isTopLayer:=false)->Result:
 	if player.health <= 0 or opponent.health <= 0:
-		var deadPlayerIs0 = (player.player_index if player.health == 0 else opponent.player_index) == 0
-
-		var itemScore: Array[float] = [0.0, 0.0]
-
-		itemScore[player.player_index] = player.sum_items()
-		itemScore[opponent.player_index] = opponent.sum_items()
+		var deadPlayerIs0 = (player.player_index if player.health <= 0 else opponent.player_index) == 0
 
 		var deathScore := [1.0, 0.0] if deadPlayerIs0 else [0.0, 1.0]
 
+		# Leftover health doesn't matter when a player dies. Set it to 0 here so CompareCurrent only compares health totals for the fraction of cases that don't end in a death.
 		var healthScore: Array[float] = [0.0, 0.0]
-		healthScore[player.player_index] = float(player.health)
-		healthScore[opponent.player_index] = float(opponent.health)
 
-		if not deadPlayerIs0:
-			if player.player_index == 0:
-				healthScore[0] += float(min(player.cigarettes, player.max_health - player.health))
-			else:
-				healthScore[0] += float(min(opponent.cigarettes, opponent.max_health - opponent.health))
+		var itemScore: Array[float] = [0.0, 0.0]
+		# The only time leftover items matter after a death is if the dealer dies in a double-or-nothing round.
```

Comment:
Setting it to 0 can have side effects after they've been multiplied further up the tree, but it might be fine with the current CompareCurrent implementation.

Reply:
The side effects with the multiplication are actually the reason I zero them out here. Imagine the dealer is evaluating two options:

Option A leads to a 50/50: 50% of the time he uses all of his remaining items to kill the player, and 50% of the time he uses no more items and doesn't kill the player.

Option B leads to a different 50/50: 50% of the time he uses half his items to kill the player, and 50% of the time he uses half his items to achieve the same outcome as the "not killing the player" outcome in option A.

Without zeroing out the item scores in the player death case, the dealer AI evaluates those two options as equal, but option B is the better option: both options have a 50% chance of killing the player, but option B conserves more items for later in the cases where the player doesn't die.